### DISCUSSION TITLE
Minor regression introduced in help functionality in `typer==0.12.2`

### DISCUSSION THREAD

**Discussion by mattkram** [fastapi/typer#795] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorials in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typer import Typer
from typer.testing import CliRunner


app = Typer(help="My cool Typer app!")


@app.callback(invoke_without_command=True, no_args_is_help=True)
def main() -> None:
    return None

# Comment below and the bug no longer occurs
#@app.command()
#def hello():
#    return None


runner = CliRunner()
result = runner.invoke(app, ["--help"])

# We expect the exit code to be 0, but it will return exit_code 1 and an exception
assert result.exit_code == 0, result.exception
```

### Description

There is a small regression introduced in #567. If a `Typer` instance is defined without any commands, but with a callback, a `ValueError` is raised when asking for the help text: `max() arg is an empty sequence`. 

I tracked the problem down to [this code](https://github.com/tiangolo/typer/pull/567/files#diff-146796270248d2c0e682c471f9a6dce7338deec90a49366cf7c2a066311af188R641-R648).

The fix is to add a check that `commands` has at least one item. If a command is added, then the issue is resolved.

I have not reviewed the source completely yet to know what a default command length should be, but I will and I am happy to contribute a fix.

### Operating System

macOS

### Typer Version

0.12.2

### Python Version

3.10.14

**Comment by svlandeg** [comment, marked as answer]
Thanks for the report and the reproducable code! This is fixed by https://github.com/tiangolo/typer/pull/796 and I used your code to create a unit test for it 🙏

**Reply by mattkram** [reply]
Thanks!

**Reply by tiangolo** [reply]
The fix is now available in Typer 0.12.3 :rocket:
